actors/builtin/miner/miner_actor.go

```diff
@@ -234,6 +242,18 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 	var recoveredSectors []*SectorOnChainInfo
 	penalty := abi.NewTokenAmount(0)
 
+	if rt.CurrEpoch()-params.ChainCommitEpoch > MaxPoStChainCommitAge {
+		rt.Abortf(exitcode.ErrIllegalArgument, "PoSt chain commit too far in the past")
+	}
+	if params.ChainCommitEpoch > rt.CurrEpoch() {
```

Comment:
This should really be >= since I don't think randomness for the current epoch is defined until the block header's for the blocks containing this messages are accumulated into a tipset and form part of chain history.

The docs for `GetRandomness` could be more clear about that though.

Reply:
any specific doc you are referring to?
if what's in here is wrong we should fix it! https://github.com/filecoin-project/specs/pull/967